tests/integration/test_extension.py

```diff
@@ -76,42 +76,72 @@ def test_fetch_global_topics(self):
                                 'topic_id': 2,
                                 'forum_id': 0,
-                                'topic_title': 'naslov teme 2',
+                                'topic_title': 'topic title 2',
                                 'topic_time': 200,
                                 'topic_first_poster_name': 'post it',
-                                'post_subject': 'tretja tema',
-                                'post_text': 'blablabla'}]),
+                                'post_subject': 'topic three',
+                                'post_text': 'hello hello'}]),
                            (3, [{
                                 'topic_id': 3,
                                 'forum_id': 0,
-                                'topic_title': 'naslov teme 3',
+                                'topic_title': 'topic title 3',
                                 'topic_time': 256,
                                 'topic_first_poster_name': 'posted it',
-                                'post_subject': 'tretja tema',
-                                'post_text': 'bla x4'}])]
-        for skip in range(0, 3):
+                                'post_subject': 'topic three',
+                                'post_text': 'hello times four'}]),
+                           (4, [{
+                                'topic_id': 4,
+                                'forum_id': 2,
+                                'topic_title': 'topic missing',
+                                'topic_time': 666,
+                                'topic_first_poster_name': 'no forum',
+                                'post_subject': 'topic yes, forum no',
+                                'post_text': 'test case'}]),
+                           (5, [{
+                               'topic_id': 5,
+                               'forum_id': 0,
+                               'topic_title': 'topic missing',
+                               'topic_time': 777,
+                               'topic_first_poster_name': 'not existent forum',
+                               'post_subject': 'topic yes, forum no',
+                               'post_text': 'test case'}]),
+                           (6, [{
+                               'unexpected_column': 5,
+                               'random_column': 0,
+                               'topic_title': 'topic missing',
+                               'topic_time': 777,
+                               'do_not_change_all': 'not existent forum',
+                               'post_subject': 'topic yes, forum no',
+                               'post_text': 'test case'}]),
+                           (7, [1, 2, 3])
+                           ]
+
+        for skip in range(0, 7):
             topic = self.app.phpbb3.fetch_global_topics(
                 skip=skip,
                 limit=1,
                 forum_id=0)
-            self.assertEqual((skip, topic), expected_topics[skip])
+            if skip > 3:
```

Comment:
I was wondering what this was for.

You can compare lists without any problems.

So, if database has positive and negative items, you only compare if actual result is equal to expected result (only positive).

Reply:
Oh, must have misunderstood the the positive vs. negative tests. All tests after index 3 are expected to not be equal.

Reply:
Yes, I see that.

What I would like to see is, that this would be more explicitly be seen from code. This does its job, but I needed cca 30 minutes longer to get the point.

By properly separating tests and naming them, you can really make the tests easy readable.

Reply:
Ok, I see what you mean, will submit new changes soon.